core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoBigQuerySupport.scala

```diff
@@ -0,0 +1,145 @@
+package org.broadinstitute.dsde.rawls.entities.datarepo
+
+import com.google.cloud.bigquery.Field.Mode
+import com.google.cloud.bigquery._
+import org.broadinstitute.dsde.rawls.entities.exceptions.{DataEntityException, EntityNotFoundException}
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+/**
+ * contains helper methods for working with BigQuery from an EntityProvider.
+ */
+trait DataRepoBigQuerySupport {
+
+  /**
+   * translate a single BigQuery FieldValue to a single Rawls AttributeValue
+   *
+   * @param field the BQ schema's field definition
+   * @param fv the BQ field value
+   * @return the Rawls attribute value, post-translation
+   */
+  def fieldValueToAttributeValue(field: Field, fv: FieldValue): AttributeValue = {
+    field.getType match {
+      case x if fv.isNull =>
+        AttributeNull
+      case LegacySQLTypeName.FLOAT | LegacySQLTypeName.INTEGER | LegacySQLTypeName.NUMERIC =>
+        AttributeNumber(fv.getNumericValue)
+      case LegacySQLTypeName.BOOLEAN =>
+        AttributeBoolean(fv.getBooleanValue)
+      case LegacySQLTypeName.STRING =>
+        AttributeString(fv.getStringValue)
+      case LegacySQLTypeName.RECORD =>
+        // TODO: unclear what to do with RECORD types; they don't translate cleanly to the entity model
```

Comment:
For now, we could ask Product if anyone will be using the Record / Struct type and what those users use it for. 
Eventually (in the Framework implementation), I think we could translate this to some kind of JSON object.

Reply:
I was thinking we could contact Jade and ask if they know of any examples of record types - and repeated values - and use that empirical knowledge to finish this implementation. I was also thinking `AttributeRawJson` might be the right answer.